src/HOFs/propMerger.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import { useImgixContext } from "../HOCs"
+
+/**
+ * Merges the `src` object into the `destination` object. Destination values are
+ * overwritten by source values. Source properties that resolve to `undefined` 
+ * are skipped if a destination value exists.
+ * 
+ * @param {Object} src - The Provider component's props object
+ * @param {Object} destination - The child component's props object
+ * @returns Object with the combined values from `src` & `destination` Objects
+ * @example
+ * let src = {foo: 'bar'}
+ * let destination = {foo: 'baz', bar: 'baz'}
+ * let result = mergeProps(src, destination)
+ * result => {foo: 'bar', bar: 'baz'}
+ * 
+ */
+export function mergeProps(src = {}, destination = {}) {
```

Comment:
I think this function needs to also recursively merge at least `imgixParams` and `htmlAttributes`, i.e.

```js
mergeProps({ imgixParams: { fit: "fill" } }, { imgixParams: { ar: "2:1" } }); 
// => { imgixParams: { fit: "fill", ar: "2:1" } },
```